### DISCUSSION TITLE
output from fractional grids

### DISCUSSION THREAD

**Discussion by JessicaMeixner-NOAA** [ufs-community/ufs-weather-model#648] [Announcements]
This is to start the conversation about having additional output when using fractional grid capabilities, so that some fields would have multiple values per grid point so that you could have a "land only temperature" and a "water only temperature" for example. 

This is also related to the need for some additional land output which is discussed in a workflow issue here: https://github.com/NOAA-EMC/global-workflow/issues/332  (@barlage @HelinWei-NOAA)

**Comment by barlage** [comment]
To be clear, as @JessicaMeixner-NOAA wrote in the global-workflow issue, there are two output related points:

1. the fractional grid outputs, this was more a desired to have a way to determine if the outputs were coming from a grid that has fractional coverage, e.g., we don't want to create land metrics/diagnostics over grids that have a mixture. For right now, we just want to be able to mask them. That's why I was hoping we could use LAND as a place to store land fraction instead of a binary flag. That is all the land evaluators need.
2. additional outputs from the land model, this has nothing to do with fractional grid; this is just adding some additional fields that will allow evaluators to do complete surface budget of energy and water.

**Reply by arunchawla-NOAA** [reply]
Would there be an impact if LAND variable is used to store land fraction instead of a binary flag? If this variable is just used internal to the code then that is easier to check out. If this variable is being output and stored then we need to do more detailed checking. 

Would the additional outputs from the land model have an impact on UPP?

**Reply by HelinWei-NOAA** [reply]
Yes. Some downstream users use this flag to determine the land points. We can output the land fraction together with the ice fraction and the water fraction  using the other name other than LAND.
> Would there be an impact if LAND variable is used to store land fraction instead of a binary flag? If this variable is just used internal to the code then that is easier to check out. If this variable is being output and stored then we need to do more detailed checking.
> 
Yes. Because we don't have GRIB definition for some of those additional outputs. 
> Would the additional outputs from the land model have an impact on UPP?

**Comment by yangfanglin** [comment]
I'd suggest not changing the definition of LAND in products.  Additionally
variables can/should be used..



On Thu, Jul 8, 2021 at 11:19 AM HelinWei-NOAA ***@***.***>
wrote:

> Yes. Some downstream users use this flag to determine the land points. We
> can output the land fraction together with the ice fraction and the water
> fraction using the other name other than LAND.
>
> Would there be an impact if LAND variable is used to store land fraction
> instead of a binary flag? If this variable is just used internal to the
> code then that is easier to check out. If this variable is being output and
> stored then we need to do more detailed checking.
>
> Yes. Because we don't have GRIB definition for some of those additional
> outputs.
>
> Would the additional outputs from the land model have an impact on UPP?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/ufs-weather-model/discussions/648#discussioncomment-980126>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKY5N2M7ZX573T74SLIIGUTTWW6Y3ANCNFSM46ZW37ZQ>
> .
>


-- 
*Fanglin Yang, Ph.D.*
*Chief, Model Physics Group*
*Modeling and Data Assimilation Branch*

*NOAA/NWS/NCEP Environmental Modeling Center*

*https://www.emc.ncep.noaa.gov/gmb/wx24fy/fyang/
<https://www.emc.ncep.noaa.gov/gmb/wx24fy/fyang/>*
